### ISSUE TITLE
Regression in prompt resetting

### ISSUE THREAD

**Issue by ajvondrak** [jeffreytse/zsh-vi-mode#219]
## General information

- Terminal program: iTerm2 Build 3.4.19
- Operating system: macOS Ventura 13.3.1 (M1 macbook running iTerm2 in Rosetta mode)
- ZSH framework: zgenom @ [`2b55d9d9`](https://github.com/jandamm/zgenom/commit/2b55d9d9c2ad8d47b299cf039cd6caa0b80abbe2)
- ZSH version: zsh 5.9 (x86_64-apple-darwin22.1.0)
- ZVM version: c4fc3a60d0c8ca9f435dd0e28cdc6ec42c4b7ad8

## Basic examination

- [x] I have read through the [README](https://github.com/jeffreytse/zsh-vi-mode) page
- [x] I have the latest version of zsh-vi-mode
- [ ] I have tested with another terminal program

## Problem description

I have a custom `PROMPT` that relies on `prompt_subst` to change based on whether I'm in Vi mode. However, upon upgrading zsh-vi-mode, this prompt stopped switching at all.

## Reproduction steps

1. Configure a custom prompt:
    ```zsh
    setopt prompt_subst
    PROMPT='$(custom_prompt)'
    custom_prompt() {
      if [[ $KEYMAP == "vicmd" ]]; then
        echo -n "vi cmd mode: "
      else
        echo -n "insert mode: "
      fi
    }
    ```
2. Hit `<esc>`/`i` to toggle between insert mode & vi cmd mode.
    - **Expect:** prompt to change between `vi cmd mode: ` and `insert mode: `
    - **Actual:** prompt stays `insert mode: `
3. Upon reverting 5979662a11e139a3a87872d1a5282c836e98e608 (and doing a `zgenom reset` and everything to make sure I'm running an up-to-date zsh-vi-mode plugin), the above test toggles the prompt properly.

## Expected behavior

I'm not sure what issue #180 was fixing (I don't use tmux), but the fix that got merged appears to cause this regression. It looks related to the prompt-resetting logic changes.

**Comment by ajvondrak** [comment]
Might be related to / a duplicate of #213?

**Comment by jeffreytse** [comment]
Hi @ajvondrak 

Thanks for your reporting. I still need some time to investigate this issue, please be patient for the update.

Thanks & Regards

**Comment by ajvondrak** [comment]
Understood & much appreciated. 🙏 In the meantime, I can work locally by
just reverting the commit in question, so I'm not really hurting too much.
Thanks for all the great work on this plugin!

**Comment by jeffreytse** [comment]
Hi @ajvondrak,

Could you please check again with the latest commit? See if your issue has been settled down. : )

Thanks & Regards

**Comment by ajvondrak** [comment]
Just tried against a3d717831c1864de8eabf20b946d66afc67e6695, and my prompt works! 🎉 

And to double-check, I went ahead and did a `git checkout HEAD^` on zsh-vi-mode a few times by hand until the prompt stopped working. Findings:
- c8ca1a1bb69e3a116c1c348d0693f849de5f645a works
- fb2f825bf01012786b9a327e5e3c2085d8d18ae5 works
- 1f28e1886dc8e49f41b817634d5c7695b6abb145 does not work

Indicating that fb2f825bf01012786b9a327e5e3c2085d8d18ae5 was indeed the first commit to fix my problem (as noted, #243 tracked the fix to this issue as well). 👍 

Thanks so much for the work here!
